### PR TITLE
Gumbel Rounding API and Testing Update

### PR DESCRIPTION
**Pull request by haihabi**
1. Add to the GPTQ configuration the Gumbel rounding config
2. Add several future testing for GPTQ

### CODE REVIEW

File: model_compression_toolkit/gptq/common/gptq_config.py [hunk 1]
```diff
@@ -68,16 +102,17 @@ def __init__(self,
...
             rho (rho): A floating point number that defines the sam optimization lookahead.
-            gumbel_entropy_regularization (float): A floating point number that defines the gumbel entropy regularization factor.
             lsb_change_per_bit_width (dict): Whether to update the bias during the training or not.
             eps (float): A floating point value for numeric stability.
             use_jac_based_weights (bool): Whether to use jacobian-based weights for weighted average loss.
             num_samples_for_loss (int): Number of samples to use for computing the jacobian-based weights.
             norm_weights (bool): Whether to normalize the returned weights (to get values between 0 and 1).
+            quantizer_config (Any): A class the contins the quantizer specific config.
+            optimizer_quantization_parameter (Any): Optimizer to override the rest optimizer  for quantizer parameters.
+            optimizer_bias (Any): Optimizer o override the rest optimizerfor bias.
```
Comment (reuvenperetz) [inline model_compression_toolkit/gptq/common/gptq_config.py:115]: Optimizer to override...

File: model_compression_toolkit/gptq/keras/quantization_facade.py [hunk 2]
```diff
@@ -47,7 +47,7 @@
 
 
     def get_keras_gptq_config(n_iter: int,
-                              optimizer: OptimizerV2 = tf.keras.optimizers.Adam(),
+                              optimizer: OptimizerV2 = tf.keras.optimizers.Adam(learning_rate=0.2),
```
Comment (reuvenperetz) [inline model_compression_toolkit/gptq/keras/quantization_facade.py:50]: Please use constants.
Also, please update the comment about the default optimizer.

File: model_compression_toolkit/gptq/common/gptq_config.py [hunk 3]
```diff
@@ -68,16 +102,17 @@ def __init__(self,
...
             sam_optimization (bool): Whether to use sam optimization.
             rounding_type (RoundingType): An enum that defines the rounding type (STE or GumbelRoudning).
             rho (rho): A floating point number that defines the sam optimization lookahead.
-            gumbel_entropy_regularization (float): A floating point number that defines the gumbel entropy regularization factor.
             lsb_change_per_bit_width (dict): Whether to update the bias during the training or not.
             eps (float): A floating point value for numeric stability.
             use_jac_based_weights (bool): Whether to use jacobian-based weights for weighted average loss.
             num_samples_for_loss (int): Number of samples to use for computing the jacobian-based weights.
             norm_weights (bool): Whether to normalize the returned weights (to get values between 0 and 1).
+            quantizer_config (Any): A class the contins the quantizer specific config.
```
[the remark this answers is not in the document]
Comment (reuvenperetz) [reply model_compression_toolkit/gptq/common/gptq_config.py:113]: Maybe we should rename it to gumbel_config?

File: model_compression_toolkit/gptq/keras/quantizer/gumbel_rounding/symmetric_gumbel.py [hunk 4]
```diff
@@ -62,9 +64,9 @@ class SymmetricGumbelRounding(GumbelRoundingBase):
     PTQ_THRESHOLD = "_ptq_threshold"
 
     def __init__(self, num_bits: int, per_axis: bool, signed: bool, power_of_two, quantization_parameter_learning: bool,
-                 temperature_learning: bool,
-                 threshold_values: np.ndarray,
-                 quantization_axis: int = -1, max_lsbs_change_map: dict = DefaultDict({}, lambda: 1),
+                 threshold_values: np.ndarray, gumbel_config: GumbelConfig,
```
Comment (reuvenperetz) [inline model_compression_toolkit/gptq/keras/quantizer/gumbel_rounding/symmetric_gumbel.py:67]: Use a new line, please

**Review by reuvenperetz — CHANGES_REQUESTED** [review]

File: model_compression_toolkit/gptq/common/gptq_config.py [hunk 5]
```diff
@@ -31,6 +36,35 @@ class RoundingType(Enum):
     GumbelRounding = 1
 
 
+class GumbelConfig(object):
+    """
```
Comment (reuvenperetz) [inline model_compression_toolkit/gptq/common/gptq_config.py:40]: Please expose this class in API and let's add it to docsrc

File: model_compression_toolkit/gptq/common/gptq_config.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (haihabi) [reply model_compression_toolkit/gptq/common/gptq_config.py:113]: No, in the future we may have a config for other quantizers.

Comment (reuvenperetz) [inline model_compression_toolkit/gptq/common/gptq_config.py:113]: Please fix the type of quantizer_config

**Review by reuvenperetz — APPROVED** [review]

**Pull request merged**